### DISCUSSION TITLE
Built-in physical button disabled if GPIO9 button1 used in Dual R2

### DISCUSSION THREAD

**Discussion by ravlar** [arendst/Tasmota#15161] [Q&A] [answered]
It's my first message and I'm not sure if I'm doing properly. Please forgive me if I'm missing any protocol/procedure. It's not my intention.

The question is about my Sonoff Dual R2. I have it with latest version (11.0.0) but the problem was there from previous versions (at least from 9.0.0). I need to use BOTH external button (configured as button1 in GPIO9) and internal built-in physical button on-board that documentation states as GPIO10. The point is that whenever I configure GPIO9 as "Button1" it behaves as expected BUT physical button suddenly stops working at all.

Things I've tried;

- Edited Template from frontend to be able to set up GPIO10 as GPIO_USER so I can manually configure behaviour. Configure module using "button1" for both GPIO9 and GPIO10. Only GPIO9 works as expected. GPIO10 on board button still disabled.
- Configured GPIO9 as "switch1" (for relay1). Everything works as expected (onboard button working and external switch working) but I really need external push button as I need to use "multi press" to control both relays with both push buttons (1 push - relay1 , 2 pushes - relay2).

Could you tell me if I'm doing anything wrong or if this is really a bug?

Best regards!

**Comment by sfromis** [comment, marked as answer]
You can only have one `Button1`, the 2nd one needs to be `Button2` instead. The index number needs to start at 1, and be unique for every occurrence of the same type.

**Reply by ravlar** [reply]
Understood! But correct me if I'm wrong... In documentation states that "multi press behaviour" is only available for button1 (default for onboard button). So how is it possible to get button1 behaviour in external push button without disabling onboard button? Is there any way to do it?

Thanks!

**Reply by sfromis** [reply]
If you want them the behave exactly the same, you can add a wire connecting them, only configuring one. 

Multi-press is available for all Buttonx pins, if the default "extra" functionality does not work for higher numbers (try it first), you can also use rules to create the actions you want.

**Reply by ravlar** [reply]
I think best way will be to directly solder external button to internal button so I finish messing up with software... I wonder why it is not possible to have same behaviour for multi press for button2, button3,...?

**Reply by sfromis** [reply]
Did you actually try it?

**Reply by ravlar** [reply]
We answered same time.  Yes, physical wire will be the way to go.

"Extra" functionality is only available for button 1 as stated in documentation:

> Button1 can directly control up to five relays. The number of the activated relay corresponds to the number of button presses and this feature is not present in the other buttons. When ButtonTopic is set to default 0 a button will always send its state for rules.

I don't think it makes sense messing with rules for such a "basic" thing but if I can't resolder push button I'll try.

**Reply by sfromis** [reply]
I do not see the word "only" in the documentation. Therefore I asked "Did you actually try it?"

It very much makes sense with rules for very basic things, arguably even more than for complex things.

**Reply by ravlar** [reply]
Mmmm ok I'll try. I guess is only using "button2" for GPIO9 isn't it?

I'll take a look at rules but first I have to "detach" buttons from relays with SetOption73, isn't it?

**Reply by sfromis** [reply]
Right, just config and see what happens :-) 

SetOption73 is one way, but it affects all button pins. You can also use rules to catch all available actions on just one Buttonx channel, this will override default logic in Tasmota.

**Reply by sfromis** [reply]
BTW.... Did a small test, and Button2 does indeed not have the multi-relay features of Button1. 

Using simple rules like `on Button2#state==11 do Power1 toggle endon` sounds like what you want, and similar for the other available button states.

**Comment by ravlar** [comment]
Perfect! Rule solution works like a charm for my needs! Best regards.
